docs/building-blocks/extending-templates/sequential-loop.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Extending Azure Resource Manager template functionality - sequential looping
+description: Describes how to extend the functionality of Azure Resource Manager templates to implement sequential looping when deploying multiple instances of a resource type.
+author: petertay
+ms.service: guidance
+ms.topic: article
+ms.date: 05/03/2017
+ms.author: pnp
+
+---
+
+# Azure Resource Manager templates - sequential looping
+
+Azure Resource Manager templates support deployment of multiple resources via a copy loop. Copy loops are typically used in the resource object to iterate over an array of variables that describe the resource. A unique name for the resource is generated by modifying the name with the copy index or iterating an array of unique strings.
+
+Copy loops are useful, but they don't work well when there are no dependencies between each member of the group. During an iteration loop, Resource Manager attempts to deploy resources in parallel. If a first resource depends on a second, the deployment may fail if Resource Manager deploys the second resource before the first resource.
+
+The real problem is that Resource Manager does not currently support `dependsOn` within an iteration loop. However, this functionality can be implemented using existing Resource Manager functionality and some creative resource naming. 
```

Comment:
what is 'dependsOn'? (Maybe just link to [here](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-define-dependencies))